mutters-core/src/main/java/com/rabidgremlin/mutters/core/AbstractSlot.java

```diff
@@ -0,0 +1,57 @@
+package com.rabidgremlin.mutters.core;
+
+import java.util.Objects;
+
+/**
+ * Base class for all slots. Slots are used to extract data from a user's utterance.
+ *
+ * @author rabidgremlin
```

Comment:
Should put your own ID here @wilmol :)

Reply:
@rabidgremlin ah it was renamed from Slot.
Tbh I prefer the abstract class approach, and think Slot should have a method ```getDefaultValue``` instead of using DefaultValueSlot interface

Reply:
We had the separation so that we can have Slots that don't have default values (or the concept of a default value)

Reply:
could have it as a hook method, returns null by default but can override if needed ?